github/config_test.go

```diff
@@ -1,82 +1,163 @@
-		// }
-		//
-		// if client := meta.(*Owner).v4client; client == nil {
-		// 	t.Fatalf("failed to return a v4 client")
-		// }
+	t.Run("returns a v3 REST API client to manage organization resources", func(t *testing.T) {
+
+		config := Config{
+			Token:        testToken,
+			BaseURL:      "https://api.github.com/",
+			Organization: testOrganization,
+			Anonymous:    false,
+		}
+		meta, err := config.Meta()
+		if err != nil {
+			t.Fatalf("failed to return meta without error: %s", err.Error())
+		}
+
+		ctx := context.Background()
+		client := meta.(*Owner).v3client
+		_, _, err = client.Organizations.Get(ctx, testOrganization)
+		if err != nil {
+			t.Fatalf("failed to validate returned client without error: %s", err.Error())
+		}
+
 	})
 
-	t.Run("returns clients configured as anonymous", func(t *testing.T) {
-		// config := Config{
-		// 	Token:      "",
-		// 	Anonymous:  true,
-		// 	Individual: true,
-		// }
-		//
-		// meta, err := config.Clients()
-		// if err != nil {
-		// 	t.Fatalf("failed to return clients without error: %s", err.Error())
-		// }
-		//
-		// if client := meta.(*Owner).v4client; client == nil {
-		// 	t.Fatalf("failed to return a v4 client")
-		// }
-		//
-		// if client := meta.(*Owner).v3client; client == nil {
-		// 	t.Fatalf("failed to return a v3 client")
-		// }
+	t.Run("returns a v4 GraphQL API client to manage organization resources", func(t *testing.T) {
+
+		config := Config{
+			Token:        testToken,
+			BaseURL:      "https://api.github.com/",
+			Organization: testOrganization,
+			Anonymous:    false,
+		}
+		meta, err := config.Meta()
+		if err != nil {
+			t.Fatalf("failed to return meta without error: %s", err.Error())
+		}
+
+		client := meta.(*Owner).v4client
+
+		var query struct {
+			Organization struct {
+				ViewerCanAdminister githubv4.Boolean
+			} `graphql:"organization(login: $login)"`
+		}
+		variables := map[string]interface{}{
+			"login": githubv4.String(testOrganization),
+		}
+		err = client.Query(context.Background(), &query, variables)
+		if err != nil {
+			t.Fatalf("failed to validate returned client without error: %s", err.Error())
+		}
+
+		if query.Organization.ViewerCanAdminister != true {
+			t.Fatalf("unexpected response when validating client")
+		}
+
 	})
 
-	t.Run("returns clients configured as individual", func(t *testing.T) {
-		// config := Config{
-		// 	Organization: "",
-		// 	Anonymous:    true,
-		// 	Individual:   true,
-		// }
-		//
-		// meta, err := config.Clients()
-		// if err != nil {
-		// 	t.Fatalf("failed to return clients without error: %s", err.Error())
-		// }
-		//
-		// if client := meta.(*Owner).v4client; client == nil {
-		// 	t.Fatalf("failed to return a v4 client")
-		// }
-		//
-		// if client := meta.(*Owner).v3client; client == nil {
-		// 	t.Fatalf("failed to return a v3 client")
-		// }
+	t.Run("manages individual resources for a GHES deployment", func(t *testing.T) {
```

Comment:
maybe add a skip condition if GHES_BASE_URL is not exported

Reply:
added in a recent commit